src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs

```diff
@@ -128,6 +128,32 @@ private static int FindIndexOfAny(string format, char[] chars, int startIndex, i
         }
 
         public string Format(object?[]? values)
+        {
+            object?[]? formattedValues = values;
+
+            if (values != null)
+            {
+                for (int i = 0; i < values.Length; i++)
+                {
+                    object formattedValue = FormatArgument(values[i]);
+                    if (!ReferenceEquals(formattedValue, values[i]))
+                    {
+                        formattedValues = new object[values.Length];
+                        Array.Copy(values, 0, formattedValues, 0, i);
+                        formattedValues[i++] = formattedValue;
+                        for (; i < values.Length; i++)
+                        {
+                            formattedValues[i] = FormatArgument(values[i]);
+                        }
+                        break;
+                    }
+                }
+            }
+
+            return string.Format(CultureInfo.InvariantCulture, _format, formattedValues ?? Array.Empty<object>());
+        }
+
+        internal string FormatQuick(object?[]? values)
```

Comment:
Should be a comment explaining when you can use FormatQuick vs Format

Reply:
Makes sense, I'll add that to the FormatQuick/FormatWithOverwrite as an xmldoc explaining that it mutates the array! Do you think it's necessary to have a note/hint about this method on the Format method as well?